models/marts/core/int_all_graph_resources.sql

```diff
@@ -29,74 +29,86 @@ unioned_with_prefix as (
+        unioned_with_calc.file_path, 
+        unioned_with_calc.directory_path,
+        unioned_with_calc.file_name,
+        case 
+            when unioned_with_calc.resource_type in ('test', 'source', 'metric', 'exposure', 'seed') then null
+            else naming_convention_prefixes.model_type 
+        end as model_type_prefix,
+        case 
+            when unioned_with_calc.resource_type in ('test', 'source', 'metric', 'exposure', 'seed') then null
+            when {{ dbt_utils.position('naming_convention_folders.folder_name_value','unioned_with_calc.directory_path') }} = 0 then null
+            else naming_convention_folders.model_type 
+        end as model_type_folder,
+        {{ dbt_utils.position('naming_convention_folders.folder_name_value','unioned_with_calc.directory_path') }} as position_folder,  
+        nullif(unioned_with_calc.column_name, '') as column_name,
+        unioned_with_calc.resource_name like 'unique%' and unioned_with_calc.resource_type = 'test' as is_not_null_test,
+        unioned_with_calc.resource_name like 'not_null%' and unioned_with_calc.resource_type = 'test' as is_unique_test,
+        unioned_with_calc.is_enabled, 
+        unioned_with_calc.materialized, 
+        unioned_with_calc.on_schema_change, 
+        unioned_with_calc.database, 
+        unioned_with_calc.schema, 
+        unioned_with_calc.package_name, 
+        unioned_with_calc.alias, 
+        unioned_with_calc.is_described, 
+        unioned_with_calc.exposure_type, 
+        unioned_with_calc.maturity, 
+        unioned_with_calc.url, 
+        unioned_with_calc.metric_type, 
+        unioned_with_calc.model, 
+        unioned_with_calc.label, 
+        unioned_with_calc.sql, 
+        unioned_with_calc.timestamp as timestamp,  
+        unioned_with_calc.source_name, -- NULL for non-source resources
+        unioned_with_calc.is_source_described, 
+        unioned_with_calc.loaded_at_field, 
+        unioned_with_calc.loader, 
+        unioned_with_calc.identifier
 
-    from unioned_with_prefix
+    from unioned_with_calc
     left join naming_convention_prefixes
-        on unioned_with_prefix.prefix = naming_convention_prefixes.prefix_value
+        on unioned_with_calc.prefix = naming_convention_prefixes.prefix_value
 
-    left join naming_convention_folders
-        -- we added '/' at the front of the path, to be /folder1/folder2/folder3/mmodel_name.sql
-        -- and we search for /naming_convention_folder/ ; this allows us to only match full folder names
-        on file_path_prefixed_with_slash like {{ dbt_utils.concat(["'%/'",'naming_convention_folders.folder_name_value',"'/%'"]) }}
-    
-    where coalesce(unioned_with_prefix.is_enabled, True) = True
-    and unioned_with_prefix.package_name != 'dbt_project_evaluator'
+    cross join naming_convention_folders   
+
+    where coalesce(unioned_with_calc.is_enabled, True) = True
+    and unioned_with_calc.package_name != 'dbt_project_evaluator'
```

Comment:
question: I am unsure of all the logic going on here, but would it make sense to filter these out before we cross join, like in the `unioned_with_calc` cte? That could save some processing time .

Reply:
sure!